src/samples/OpenChannelMessageThreading.js

```diff
@@ -0,0 +1,718 @@
+        updateMessage={updateMessage}
+        threadParentsMessage={state.threadParentsMessage}
+      >
+        <MessagesList
+          displayNone={"display-none"}
+          messages={state.threadMessages}
+          handleDeleteMessage={handleDeleteMessage}
+          updateMessage={updateMessage}
+        />
+        <MessageInput
+          threadInputClass={"thread-input"}
+          value={state.threadMessageInputValue}
+          isOpenThread={state.isOpenThread}
+          isThread={true}
+          onChange={onThreadMessageInputChange}
+          sendMessage={sendThreadMessage}
+          fileSelected={state.threadFile}
+          onFileThreadInputChange={onFileThreadInputChange} />
+      </Thread>
+    </>
+  );
+};
+
+// Chat UI Components
+const ChannelList = ({ channels, handleJoinChannel, toggleShowCreateChannel, handleDeleteChannel, toggleChannelDetails }) => {
+  return (
+    <div className='channel-list'>
+      <div className="channel-type">
+        <h1>Open Channels</h1>
+        <button className="channel-create-button" onClick={toggleShowCreateChannel}>Create Channel</button>
+      </div>
+      {
+        channels.map(channel => {
+          const userIsOperator = channel.operators.some((operator) => operator.userId === sb.currentUser.userId)
+          return (
+            <div key={channel.url} className="channel-list-item" >
+              <div className="channel-list-item-name"
+                   onClick={() => { handleJoinChannel(channel.url) }}>
+                {channel.name}
+              </div>
+              {userIsOperator &&
+              <div>
+                <button className="control-button" onClick={() => toggleChannelDetails(channel)}>
+                  <img className="channel-icon" src='/icon_edit.png' />
+
+                </button>
+                <button className="control-button" onClick={() => handleDeleteChannel(channel.url)}>
+                  <img className="channel-icon" src='/icon_delete.png' />
+
+                </button>
+              </div>}
+            </div>);
+        })
+      }
+    </div >);
+}
+
+
+const Channel = ({ currentlyJoinedChannel, handleLeaveChannel, children }) => {
+  if (currentlyJoinedChannel) {
+    return <div className="channel">
+      <ChannelHeader>{currentlyJoinedChannel.name}</ChannelHeader>
+      <div>
+        <button className="leave-channel" onClick={handleLeaveChannel}>Exit Channel</button>
+      </div>
+      <div>{children}</div>
+    </div>;
+
+  }
+  return <div className="channel"></div>;
+
+}
+
+const Thread = ({ isOpenThread, exitThread, children, threadParentsMessage, handleDeleteMessage, updateMessage }) => {
+  return isOpenThread && (
+    <div className="channel thread">
+      <ChannelHeader>Thread</ChannelHeader>
+      <div>
+        <button className="leave-channel" onClick={() => exitThread()}>Exit Thread</button>
+      </div>
+      <Message
+        displayNone={"display-none"}
```

Comment:
Passing around a specific display none prop seem a bit much here. Can we not infere when to display the thread based on a property in the state object. perhaps threadMessages.